tensorboard/webapp/metrics/views/card_renderer/scalar_card_component.ts

```diff
@@ -163,27 +163,34 @@ export class ScalarCardComponent<Downloader> {
 
   getCursorAwareTooltipData(
     tooltipData: TooltipDatum<ScalarCardSeriesMetadata>[],
-    cursorLoc: {x: number; y: number}
+    cursorLocationInDataCoord: {x: number; y: number},
+    cursorLocation: {x: number; y: number}
   ): ScalarTooltipDatum[] {
     const scalarTooltipData = tooltipData.map((datum) => {
       return {
         ...datum,
         metadata: {
           ...datum.metadata,
           closest: false,
-          distSqToCursor: Math.hypot(
-            datum.point.x - cursorLoc.x,
-            datum.point.y - cursorLoc.y
+          distToCursor: Math.hypot(
+            datum.point.x - cursorLocationInDataCoord.x,
+            datum.point.y - cursorLocationInDataCoord.y
           ),
+          distToCursorPixels: Math.hypot(
+            datum.pixelLocation.x - cursorLocation.x,
+            datum.pixelLocation.y - cursorLocation.y
+          ),
+          distToCursorX: datum.point.x - cursorLocationInDataCoord.x,
+          distToCursorY: datum.point.y - cursorLocationInDataCoord.y,
         },
       };
     });
 
     let minDist = Infinity;
     let minIndex = 0;
     for (let index = 0; index < scalarTooltipData.length; index++) {
-      if (minDist > scalarTooltipData[index].metadata.distSqToCursor) {
-        minDist = scalarTooltipData[index].metadata.distSqToCursor;
+      if (minDist > scalarTooltipData[index].metadata.distToCursor) {
```

Comment:
I don't know (haven't looked at the code in detail) but suspect this is calculating the "closest" point to the cursor which is then highlighted in the tooltip UI.

![image](https://user-images.githubusercontent.com/17152369/210593947-21e6b4cd-38e0-4ee5-a96f-23c59003c994.png)

Does this need to change depending on which sorting method is chosen?

Reply:
This does need to be updated but I'm not sure it need to change. Given that we are changing the definition of "nearest", I vote to just change this to use the new definition.

Reply:
Ok I think that makes sense. Thanks.